tests/hash-backend-integration/src/tests/subgraph/friendship.test.ts

```diff
@@ -0,0 +1,184 @@
+  const graphContext = createTestImpureGraphContext();
+
+  it("can upload snapshot", async () => {
+    await expect(
+      restoreSnapshot(path.join(__dirname, "pass", "friendship.jsonl")),
+    ).resolves.not.toThrowError();
+  });
+
+  it("read all entities", async () => {
+    const entities = await getEntities(graphContext, {
+      query: {
+        filter: {
+          all: [],
+        },
+        graphResolveDepths: zeroedGraphResolveDepths,
+        temporalAxes: fullDecisionTimeAxis,
+      },
+    }).then(getRoots);
+
+    expect(entities.length).toBe(5);
+  });
+
+  it("read entities at 2000-01-01 as of now", async () => {
+    const entities = await getEntities(graphContext, {
+      query: {
+        filter: {
+          all: [],
+        },
+        graphResolveDepths: zeroedGraphResolveDepths,
+        temporalAxes: temporalAxesForTimestamp(
+          "2000-01-01T00:00Z" as Timestamp,
+          null,
+        ),
+      },
+    }).then(getRoots);
+
+    expect(entities.length).toBe(0);
+  });
+
+  it("read entities at 2001-01-01 as of now", async () => {
+    const entities = await getEntities(graphContext, {
+      query: {
+        filter: {
+          all: [],
+        },
+        graphResolveDepths: zeroedGraphResolveDepths,
+        temporalAxes: temporalAxesForTimestamp(
+          "2001-01-01T00:00Z" as Timestamp,
+          null,
+        ),
+      },
+    }).then(getRoots);
+
+    expect(entities.length).toBe(1);
+  });
+
+  it("read entities at 2001-01-01 as of 2001-01-01", async () => {
+    const entities = await getEntities(graphContext, {
+      query: {
+        filter: {
+          all: [],
+        },
+        graphResolveDepths: zeroedGraphResolveDepths,
+        temporalAxes: temporalAxesForTimestamp(
+          "2001-01-01T00:00Z" as Timestamp,
+          "2001-01-01T00:00Z" as Timestamp,
+        ),
+      },
+    }).then(getRoots);
+
+    expect(entities.length).toBe(0);
+  });
+
+  it("read entities at 2001-01-01 as of 2001-01-20", async () => {
+    const entities = await getEntities(graphContext, {
+      query: {
+        filter: {
+          all: [],
+        },
+        graphResolveDepths: zeroedGraphResolveDepths,
+        temporalAxes: temporalAxesForTimestamp(
+          "2001-01-01T00:00Z" as Timestamp,
+          "2001-01-20T00:00Z" as Timestamp,
+        ),
+      },
+    }).then(getRoots);
+
+    expect(entities.length).toBe(1);
+  });
+
+  it("read entities at 2002-01-01 as of 2001-01-01", async () => {
+    const entities = await getEntities(graphContext, {
+      query: {
+        filter: {
+          all: [],
+        },
+        graphResolveDepths: zeroedGraphResolveDepths,
+        temporalAxes: temporalAxesForTimestamp(
+          "2002-01-01T00:00Z" as Timestamp,
+          "2001-01-01T00:00Z" as Timestamp,
+        ),
+      },
+    }).then(getRoots);
+
+    expect(entities.length).toBe(0);
+  });
+
+  it("read entities at 2002-02-01 as of now", async () => {
```

Comment:
Having these "readable" date in the test data snapshot seems to have required quite a bit of manual tuning. Is this something we're going to do for all test data? It could be quite time intensive for us to clean the data. (similar question for the UUIDs). Or is it mostly just because this test snapshot is so "small" so it was easy to do?

Reply:
I was thinking about writing a script for this but for now I just did it with this data set as it's small enough